### ISSUE TITLE
refactor: remove outdated code for backwards compatibility

### ISSUE THREAD

**Issue by ravinderk** [impress-org/give#3003]
@DevinWalker This is information about backward compatibility code in core: 

- [x] CMB2 Backward compatibility ( since 1.8 ): 
   https://github.com/impress-org/give/blob/master/includes/admin/settings/class-settings-cmb2-backward-compatibility.php `give_render_field`
    License setting page still renders with cmb2 hook, we can remove backward compatibility and update this code.

- [x] Give Setting ( since 1.8 ):
    https://github.com/impress-org/give/blob/master/includes/admin/class-give-settings.php
    We do not need this setting class, so we can remove this file and relocate helper function if any.
    We also removed the setting, so we can also drop compatibility for old setting names.
- [x] Form Setting ( since 1.8 ):
      https://github.com/impress-org/give/blob/07ca0a3a95f8d7f38c23b19d38987311730d4b30/includes/admin/give-metabox-functions.php#L1384
       We can drop backward compatibility for old form settings
- [x] Setting Filters ( since 1.8 ):
       Each filter consume too much time to set up, so we can remove backward compatibility for an old setting filter like: `give_settings_licenses`

**Comment by DevinWalker** [comment]
@ravinderk I think it's a good idea that we clean out old backwards compatibly code but can you please provide more description on this issue?

1. Code pre-2.0? Or another version?
2. CMB code? If so, that would be great.
3. Where the majority of the backwards compatibility code lives (file names, line numbers).

**Comment by ravinderk** [comment]
@DevinWalker This is information about backward compatibility code in core: 
- [ ] CMB2 Backward compatibility ( since 1.8 ): 
   https://github.com/impress-org/give/blob/master/includes/admin/settings/class-settings-cmb2-backward-compatibility.php `give_render_field`
    License setting page still renders with cmb2 hook, we can remove backward compatibility and update this code.

- [ ] Give Setting ( since 1.8 ):
    https://github.com/impress-org/give/blob/release/2.4.0/includes/admin/class-give-settings.php
    We do not need only setting class, so we can remove this file and relocate helper function if any.
    We also removed the setting, so we can also drop compatibility for old setting names.

- [ ] Payment Meta ( since 2.0 ):
    https://github.com/impress-org/give/blob/release/2.4.0/includes/payments/backward-compatibility.php
   We have backward compatibility for old payment meta key. It can break addons, so we can wait till two more major release to remove this.

- [ ] Donor Meta (since 2.0 ):
    https://github.com/impress-org/give/blob/release/2.4.0/includes/donors/backward-compatibility.php
    We have backward comaptibility for donor meta key `_give_user_address` and we can remove this.

- [ ] Email (since 2.0 ):
    https://github.com/impress-org/give/blob/release/2.4.0/includes/admin/emails/backward-compatibility.php
    We can remove old email setting backward compatibility. 
- [ ] Comment (since 2.2.0):
      https://github.com/impress-org/give/blob/release/2.4.0/includes/class-give-comment.php
      We can wait on this one to delete.

- [ ] Remove backward compatibility for old `RIAL` currency:
       https://github.com/impress-org/give/blob/release/2.4.0/includes/admin/tools/export/pdf-reports.php#L68
- [ ] Form Setting ( since 1.8 ):
       https://github.com/impress-org/give/blob/release/2.4.0/includes/admin/give-metabox-functions.php#L1384
       We can drop backwardc compatibility for old form settings

- [ ] Donor Class (since 1.6 ):
       `Give_DB_Customers, Give_DB_Customer_Meta`
      We can remove compatibility for old donor db related classes.

- [ ] Setting Filters ( since 1.8 ):
       Each filter consume too much time to set up, so we can remove backward compatibility for an old setting filter like: `give_settings_licenses`

 - [ ]  Log ( since 2.0 ):
       Previously we were storing logs into posts table with log post type, but now we have our own table, so we can remove backward compatibility for the log post type.

- [ ] Email Tags ( since 2.0 ):
       We updated email tag render function to accept array instead of a string, but we have backward compatibility for old argument. We can take little more time to remove this backward compatibility.

- [ ] Old Data (since 2.0):
      We still have old data in postmeta, posts, and other table but we can take a little more time to delete that data.

**Comment by ravinderk** [comment]
WIP in https://github.com/impress-org/give/tree/issue/3003

**Comment by kevinwhoffman** [comment]
Moving this back to Product Backlog so that we can address the remaining 2.0+ items at a later date.

**Comment by DevinWalker** [comment]
We decided to focus `2.5.0` deprecation on `>2.0.0` backwards compatibility items. We are creating a new issue for the `<2.0.0` items and will focus on those once more people have upgraded.
